src/MX/MegaMenu/Controller/Adminhtml/Menu/ExportPost.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace MX\MegaMenu\Controller\Adminhtml\Menu;
+
+use MX\MegaMenu\Controller\Adminhtml\Menu as MenuController;
+use MX\MegaMenu\Model\MenuFactory;
+use MX\MegaMenu\Api\MenuRepositoryInterface;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Registry;
+
+class ExportPost extends MenuController
+{
+    /**
+     * @var MenuFactory
+     */
+    private $menuFactory;
+
+    /**
+     * @var MenuRepositoryInterface
+     */
+    private $menuRepository;
+
+    /**
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param MenuFactory|null $menuFactory
+     * @param MenuRepositoryInterface|null $menuRepository
+     */
+    public function __construct(
+        Context $context,
+        Registry $coreRegistry,
+        MenuFactory $menuFactory = null,
+        MenuRepositoryInterface $menuRepository = null
+    ) {
+        $this->menuFactory = $menuFactory ?: ObjectManager::getInstance()->get(MenuFactory::class);
+        $this->menuRepository = $menuRepository ?: ObjectManager::getInstance()->get(MenuRepositoryInterface::class);
+
+        parent::__construct($context, $coreRegistry);
+    }
+
+    public function execute()
+    {
+        $response = [
+            'status' => false,
+            'result' => '',
+            'redirect' => $this->getUrl('mx_megamenu/menu/export')
+        ];
+
+        try {
+            $result = [];
+            $items = $this->menuRepository->getAllItems();
+
+            if ($items->count()) {
```

Comment:
no need to check the count since the foreach will run 0 times anyway if the $items is an empty collection/array